lighthouse-core/lib/network-recorder.js

```diff
@@ -327,6 +327,37 @@ class NetworkRecorder extends EventEmitter {
     return request;
   }
 
+  /**
+   * @param {NetworkRequest} record The record to find the initiator of
+   * @param {Map<string, NetworkRequest>} Network requests that are possibly prefetch requests,
+   *     keyed by URL.
+   * @param {Map<string, NetworkRequest>} Network requests that are definitely not prefetch
+   *     requests, keyed by URL.
+   * @param {?NetworkRequest} The request that initiated record, or null if no valid initiator
+   *     was found.
+   *@private
+   */
+  static _chooseInitiatorRequest(record, possiblePrefetchRecords, nonPrefetchRecords) {
+    if (record.redirectSource && record.redirectSource.responseReceivedTime <= record.startTime) {
+      return record.redirectSource;
+    }
+    const stackFrames = (record.initiator.stack && record.initiator.stack.callFrames) || [];
+    const initiatorURL = record.initiator.url || (stackFrames[0] && stackFrames[0].url);
+    // Choose a non-prefetch request if one exists.
+    if (nonPrefetchRecords.has(initiatorURL) &&
```

Comment:
nit: I think this flows a little better if it's something like
```js
const nonPrefetchRequest = nonPrefetchRecords.get(initiatorURL);
if (nonPrefetchRequest && nonPrefetchRequest.responseReceivedTime <= record.startTime) {
  return nonPrefetchRequest;
}
```
 same with below

Reply:
Yeah, that's much better!
I ended up refactoring the code so it's no longer relevant